File: apps/backend/src/EnterpriseAgentOs.Domain/features/Mcp/Interfaces/IAgentMcpServerRepository.cs
namespace EnterpriseAgentOs.Domain.Features.Agents.Integrations;

public interface IAgentIntegrationRepository
{
    Task<IReadOnlyList<string>> ListIntegrationNamesForAgentAsync(Guid agentId, CancellationToken ct = default);
    Task AssignAsync(Guid agentId, string integrationName, CancellationToken ct = default);
    Task UnassignAsync(Guid agentId, string integrationName, CancellationToken ct = default);
    Task UnassignIntegrationFromAllAgentsAsync(string integrationName, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/mcp/IAgentMcpServerRepository.java
package com.enterpriseagentos.backendjava.domain.features.mcp;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public interface IAgentMcpServerRepository {
    CompletableFuture<List<String>> listServerNamesForAgentAsync(UUID agentId);
    CompletableFuture<Void> assignAsync(UUID agentId, String mcpServerName);
    CompletableFuture<Void> unassignAsync(UUID agentId, String mcpServerName);
}
